public/scripts/preset-manager.js

```diff
@@ -182,6 +183,10 @@ class PresetManager {
     }
 
     async savePreset(name, settings) {
+        if (this.apiId === 'instruct') {
+            await checkForSystemPromptInInstructTemplate(name, settings);
+        }
+
         const preset = settings ?? this.getPresetSettings(name);
```

Comment:
### 1. Crashes On Saving / Save As for Instruct
Crashes because same `savePreset` calls don't input the settings. jsDocs would've actually helped your to not miss that, if it was marked as optional :P

![image](https://github.com/user-attachments/assets/839de11d-77bf-489d-8e0f-6efc7281d980)

Fix is easy, switch the lines
```suggestion
        const preset = settings ?? this.getPresetSettings(name);
        
        if (this.apiId === 'instruct') {
            await checkForSystemPromptInInstructTemplate(name, preset);
        }
```